### ISSUE TITLE
Create `Admin:Advanced` permission for Advanced Admins

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-server#7249]
Part of #7176.

Create an `Admin:Advanced` permission, which will be used to gate Advanced Admin tools (per the [PRD](https://docs.google.com/document/d/1ts_1IB3hXcTvEX0gclN55NsBVs7r1duBOM9zQaqSyhE/edit)). ~~Assign this permission to the `System Administrators` groups, which will be used to determine who are Full Admins.~~

**Comment by diox** [comment]
We already have an `Admin:%` permission that only the `Admins` group should have. Can we re-use this (and rename the group) ?

**Comment by jvillalobos** [comment]
The goal here is to limit what Full Admins do, so they don't have permission to do whatever and then we are safer including more people in that group. The current admin group is more like "System Admins", and we want to have a lower access group that is more like "Site Admins". Hope this makes sense :).

**Comment by eviljeff** [comment]
I got: All:Admin is full admin is system administrators but it will "be used to gate Full Admin tools" and "to limit what Full Admins do".  So full != full and system administrator is not someone who can administrator the system (i.e. has `*:*`)?  Doesn't make sense to me, sorry.  

The PRD doesn't mention System Admins or Site Admins, only Full Admins.  Has the [4 user types](https://docs.google.com/document/d/1ts_1IB3hXcTvEX0gclN55NsBVs7r1duBOM9zQaqSyhE/edit#heading=h.74vwcf1wl21) been expanded to 5?

**Comment by jvillalobos** [comment]
Sorry, terminology is my eternal foe.

The PRD defines "Full Admins" as the people who will be able to use all of the admin tools specified in the doc. They won't necessarily have access to `*:*`, which is what I referred to as "System Admins" in my previous comment. The goal is that we can have more people as Full Admins without worrying that they have access to absolutely everything.

To put it differently, System Admins > Full Admins > (Admin Reviewer, Content Curator, Staff).

I can change the name if that makes sense.

**Comment by diox** [comment]
Is there a table somewhere with all admin-related groups showing Group Name - Tools they need access to ? With that we could figure out exactly what we want.

**Comment by diox** [comment]
(Ignoring what group currently exist if that helps - we can change them as needed)

**Comment by jvillalobos** [comment]
**System Admins** (not mentioned in PRD)
Permission: `*:*`
Access to: all admin tools in PRD, everything under `/admin/`, including `/admin/models/`

**Full Admins**
Permission: `All:Admin`
Access to: admin tools in PRD labeled as for "Full Admins".

**Other Admin Groups**
Similar to Full Admins, they should each have their own permission and access limited to the tools specified in the PRD. 

The `All:Admin` permission shouldn't give access to other tools. For example, if you only have `All:Admin`, you shouldn't be able to access tools that are for "Curators". We'll assign users to multiple user groups to make sure they have all the access they need.

`*:*` is the only permission that should grant access to everything.

**Comment by jvillalobos** [comment]
> so ideally we'd use a different name

Okay, to try to clarify this, I've renamed things (in the PRD and the opening comment of this issue). Full Admin -> Advanced Admin, and `All:Admin` -> `Admin: Advanced`.

> Do "Full Admins" actually have full access to admin tools?

The `Admin: Advanced` permission should only gate features that are indicated as for Advanced Admins in the PRD. I want each feature to be gated by a single permission, and then we can use group mappings to give users the permissions they need. For Advanced Admins, I think what we should do is give that group `Admin: Advanced` plus the other admin permissions (Content curator and Admin Reviewer).

> Is there a simple list of the admin tools for full admins? And/or a complete list of tools?

The PRD has a full list of the tools in the form of user stories. #7176 points to which tools are planned for the first phase. I can provide them in a different format if that's useful, but I'll need to understand what info is missing or considered unnecessary.

> it would be more consistent to have "Admin:Curate", "Admin:ReplacementAddons" "Admin:LanguageTools", etc.

The problem with that approach is that the list of permissions grows very quickly. It's not that there's a direct mapping between whole tools and permissions, but rather specific actions in certain tools that only Advanced Admins should be able to do.

For example, the user admin tool. The basic admin (staff) has essentially read-only access. Admin reviewers can ban users. Advanced admins can delete users. If we had multiple permissions per tool, I think managing groups will become fairly complicated.

**Comment by diox** [comment]
Note: because it's a nice summary, I've edited https://github.com/mozilla/addons-server/issues/7249#issuecomment-375784344 to reflect those terminology changes.

**Comment by diox** [comment]
Going to use this issue to create the permission and a group that has it, but the implementation will be in separate issues. For instance, #7268 will use it for user deletion.

**Comment by AlexandraMoga** [comment]
I have added a user to the Advanced Admin group on AMO stage and then accessed the Admin Tools with that account.
Note: not all actions mentioned in the PRD are currently implemented. 
- here are the Tools this user can access in `/admin/models/`
*Note: they can only _view_ contents from Replacement addons


![image](https://user-images.githubusercontent.com/31961530/43512658-577b9f98-9584-11e8-831c-70343c3221ab.png)
- they can edit add-ons from Admin Tools or directly from DevHub
- they can edit user profiles from Admin Tools or from the frontend
- they can edit collection metadata and delete a collection from Admin tools

@diox, when I've tried to delete a user that had associated add-ons, ratings, collections, abuse reprots I have received the following message:
 

![image](https://user-images.githubusercontent.com/31961530/43514434-dd98e5dc-9588-11e8-84af-cd1188ea873b.png)


Let me know if this is expected for Advanced Admins. If not, I can open a follow up issue.
